readthedocs/projects/notifications.py

```diff
@@ -11,3 +11,10 @@ class ResourceUsageNotification(Notification):
     context_object_name = 'project'
     subject = 'Builds for {{ project.name }} are using too many resources'
     level = REQUIREMENT
+
+
+class DeprecatedWebhookEndpointNotification(Notification):
+    name = 'deprecated_webhook_endpoint'
+    context_object_name = 'project'
+    subject = 'Project {{ project.name }} is using a deprecated webhook'
+    level = REQUIREMENT
```

Comment:
Noted earlier, but our notification system was put into place primarily for this use case. See our fallback storage backend in `readthedocs.notifications.storages`. Setting `PERSISENT_REQUIREMENT` level here should 1) make the notification sticky and 2) will handle message deduplication for us.

Reply:
I don't see `PERSISTENT_REQUIREMENT` level. `REQUIREMENT` is then changed to `WARNING_PERSISTENT` which I think it's what we want.

I think that my current implementation covers 1) and 2). The problem is when/how send the email. That's what I mentioned at https://github.com/rtfd/readthedocs.org/pull/4898#discussion_r232987110 and why I want to send them with a celery task at https://github.com/rtfd/readthedocs.org/pull/4898#discussion_r233086804

If you already have in mind the proper implementation for this with the current code, I'm not really following you :(

Reply:
Ah nevermind, it's the `SiteNotification` backend that converts `REQUIREMENT` to `PERSISENT_WARNING` via the mapping. It doesn't need to be set as `PERSISENT_WARNING` on the notification subclass.